src/d3a/models/strategy/storage.py

```diff
@@ -284,10 +284,10 @@ def buy_energy(self, market):
                 # Don't buy our own offer
                 continue
 
-            # TODO: find a better was to distinguish between IAA and devices
-            if "IAA" in offer.seller \
-                    and ConstSettings.IAASettings.AlternativePricing.PRICING_SCHEME != 0:
-                # dont buy from IAA if alternative pricing scheme is activated
+            alt_pricing_settings = ConstSettings.IAASettings.AlternativePricing
+            if offer.seller == alt_pricing_settings.ALT_PRICING_MARKET_MAKER_NAME \
+                    and alt_pricing_settings.PRICING_SCHEME != 0:
+                # don't buy from IAA if alternative pricing scheme is activated
```

Comment:
if alt_pricing_settings.PRICING_SCHEME != 0:
Shouldn't this be enough?

Reply:
It is a safety precaution, for the storage to never buy energy from the IAA, it is not necessary per se, but I feel safer to have it here.

Reply:
Then I guess it should be with or. Right?
            if offer.seller == alt_pricing_settings.ALT_PRICING_MARKET_MAKER_NAME \
                    or alt_pricing_settings.PRICING_SCHEME != 0:

Reply:
No, the `offer.seller == alt_pricing_settings.ALT_PRICING_MARKET_MAKER_NAME` constraint should only be applied if an alternative pricing scheme is selected. Otherwise you disable buying from IAA for all cases.